include/deal.II/matrix_free/tools.h

```diff
@@ -98,6 +98,64 @@ namespace MatrixFreeTools
     const unsigned int quad_no                  = 0,
     const unsigned int first_selected_component = 0);
 
+  /**
+   * Compute the diagonal of a linear operator (@p diagonal_global), given
+   * @p matrix_free and the local cell integral operation @p local_vmult. The
+   * vector is initialized to the right size in the function.
```

Comment:
Aren't we assuming that the user should initialize the vector nowadays?

Reply:
You are right. The comment was still at the old function. I have removed it there too.